File: src/main/scala/com/github/mauricio/postgresql/parsers/NoticeParser.scala
package com.github.mauricio.postgresql.parsers

import com.github.mauricio.postgresql.messages.backend.{NoticeMessage, Message}
import java.nio.charset.Charset

/**
 * User: Maurício Linhares
 * Date: 3/1/12
 * Time: 10:06 PM
 */

class NoticeParser( charset : Charset ) extends InformationParser(charset) {

  def createMessage(fields: Map[Char, String]): Message = new NoticeMessage(fields)

}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/NoticeParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.NoticeMessage
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import java.nio.charset.Charset

class NoticeParser(charset: Charset) : InformationParser(charset) {
  fun createMessage(fields: Map<Char, String>): ServerMessage = NoticeMessage(fields)
}
